Diff between wp-fork branch and main repo's default branch

ceyhun opened the pull request:
## Bump React Native version (updated by @fluiddot )
Check the following instructions for updating the RN version: https://github.com/wordpress-mobile/react-native-reanimated/pull/3#issuecomment-1054378430

## Tweaks applied to the fork

- GitHub workflow
  - Point to `trunk` branch instead of `main`.
- Podspec file
  - Set React Native version manually and remove logic for calculating.
  - Change `source` to point to the forked repository.
- `android-npm`
  - Add `build.gradle` file to publish the Android binaries.
- `android/build.gradle`
  - Update the Gradle plugin version.
  - Update Sdk versions.
  - Change repositories to use our React Native mirror Maven repository (only when publishing the binaries).
  - Update dependencies to use a specific React Native (only when publishing the binaries).
- `android/gradle.properties`
  - Update settings for Jitpack
- Gradle wrapper
  - Use a different version.
- `android/settings.gradle`
  - Use Kotlin version.
- `createNPMPackage` script
  - Update the script to use a specific React Native version and enable Hermes.
- Add Jitpack configuration.
- Add Jest utils patch (probably could be removed in the future).

jgonet committed:
Release 1.9.0

dratwas committed:
add jitpack setup

dratwas committed:
fix repositories for local build

hypest committed:
Bintray is no more, use our S3 mirror

hypest committed:
Bump RN binary to 0.64.0

ceyhun committed:
Merge pull request #2 from wordpress-mobile/feat/upgrade-rn-binary-to-0.64.0

ceyhun committed:
Point podspec source to fork repo

ceyhun committed:
Add -wp to version number

oguzkocer committed:
Upgrade Gradle to 7.1.1

oguzkocer committed:
Use plugin DSL

oguzkocer committed:
Simplify android extension

oguzkocer committed:
Update repositories & dependencies

oguzkocer committed:
Update gradle.properties

oguzkocer committed:
Adds maven-publish plugin

oguzkocer committed:
Remove jitpack.yml

oguzkocer committed:
Merge pull request #4 from wordpress-mobile/upgrade-to-gradle-7.1.1

Upgrade to gradle 7.1.1

piaskowyk committed:
Added support for hermes on rn65

piaskowyk committed:
Gradle update

piaskowyk committed:
Update iOS init

piaskowyk committed:
Update cmake

piaskowyk committed:
Add maven repo

piaskowyk committed:
Add maven repo v2

piaskowyk committed:
Add gradle from master

piaskowyk committed:
Update package.json

piaskowyk committed:
Changes for new Hermes

piaskowyk committed:
Remove jni refference for iOS

piaskowyk committed:
change to unque ptr

piaskowyk committed:
Fix problem with user store

piaskowyk committed:
Removed pickFirst

piaskowyk committed:
Update packae.json

piaskowyk committed:
Update package.json

piaskowyk committed:
Remove unused .so from package

ceyhun committed:
Use correct version number in package.json

ceyhun committed:
Add tarball file

fluiddot committed:
Merge pull request #5 from wordpress-mobile/try/tarball

Add tarball

piaskowyk committed:
Support for RN66

piaskowyk committed:
Release 2.2.4

piaskowyk committed:
Update ersion in package.json

dcalhoun committed:
Update to RN 0.66.2

Routine RN package upgrade.

dcalhoun committed:
Bump version

dcalhoun committed:
Generate new tarball

dcalhoun committed:
Merge pull request #6 from wordpress-mobile/chore/upgrade-to-react-native-0.66

Update to RN 0.66.2

oguzkocer committed:
Update CI branch from master to trunk

AliSoftware committed:
Merge pull request #7 from wordpress-mobile/update/ci-branch

Update CI branch from master to trunk

wfolini committed:
Add reactInstanceManager getter (#2863)

## Description
This PR adds a getter for `ReactInstanceManager`. The motivation for this PR is to fix issues #2719 and #2801

Those Android apps whose application class does not implement `ReactApplication` or simply have a different mechanism for storing a `ReactInstanceManager`, currently they have an incompatibility with `react-native-reanimated` ending in a crash when launching the app, as the issues indicates.
Normally, those apps are where [React Native is integrated with existing Android apps](https://reactnative.dev/docs/integration-with-existing-apps)

So, introducing this getter allows us to override this getter and implement a custom way to return the `ReactInstanceManager` to be used by `react-native-reanimated`.

<!--
Description and motivation for this PR.

Inlude Fixes #<number> if this is fixing some issue.

Fixes # .
-->

## Changes
- Added `getReactInstanceManager` method for android.
<!--
Please describe things you've changed here, make a **high level** overview, if change is simple you can omit this section.

For example:

- Added `foo` method which add bouncing animation
- Updated `about.md` docs
- Added caching in CI builds

-->

<!--

## Screenshots / GIFs

Here you can add screenshots / GIFs documenting your change.

You can add before / after section if you're changing some behavior.

### Before

### After

-->

## Test code and steps to reproduce
This is an example of how to use and override this getter. It is necessary to manually link `react-native-reanimated` before to be able to do it.
```java

public class MainActivity extends Activity implements DefaultHardwareBackBtnHandler {

    private static ReactInstanceManager mReactInstanceManager;
    ...

    @Override
    protected void onCreate(Bundle savedInstanceState) {
       ...
        List<ReactPackage> packages = new PackageList(getApplication()).getPackages();

        // Adding manually Reanimated package here, with overriding getReactInstanceManager method
         packages.add(new ReanimatedPackage() {
            @Override
            public ReactInstanceManager getReactInstanceManager(ReactApplicationContext reactContext) {
               // Implement here your way to get the ReactInstanceManager
               return MainActivity.getReactInstanceManager();
            }
         });

        mReactInstanceManager = ReactInstanceManager.builder()
                .setApplication(getApplication())
                .setCurrentActivity(this)
                .setBundleAssetName("index.android.bundle")
                .setJSMainModulePath("index")
                .setJSIModulesPackage(new ReanimatedJSIModulePackage()) // Adding ReanimatedJSIModulePackage here
                .addPackages(packages)
                .setUseDeveloperSupport(BuildConfig.DEBUG)
                .setInitialLifecycleState(LifecycleState.RESUMED)
                .build();
       
        ...
    }
...
}
```
<!--
Please include code that can be used to test this change and short description how this example should work.
This snippet should be as minimal as possible and ready to be pasted into editor (don't exclude exports or remove "not important" parts of reproduction example)
-->

## Checklist

- [x] Included code example that can be used to test this change
- [ ] Updated TS types
- [ ] Added TS types tests
- [ ] Added unit / integration tests
- [ ] Updated documentation
- [ ] Ensured that CI passes

fluiddot committed:
Update react-native version in package.json

fluiddot committed:
Add Android project for publishing binary

fluiddot committed:
Merge branch 'upstream-2.2.4' into wp-fork-2.2.4

# Conflicts:
#	android/build.gradle
#	package.json

fluiddot committed:
Add wp-1 suffix to package version

fluiddot committed:
Generate tarball

fluiddot committed:
Generate tarball

fluiddot committed:
Update yarn.lock

fluiddot committed:
create npm package only for a specific RN version

fluiddot committed:
Simplify android-npm build configuration

The original configuration tries to fetch the RN version of the app and find a matching AAR. In our case, since we control the RN version, we can simplify the configuration by providing a single and specific RN version.

fluiddot committed:
Generate AAR for RN version 0.66.2

fluiddot committed:
Generate tarball

fluiddot committed:
Remove maven-publish project

fluiddot committed:
Specify react native version in podspec

fluiddot committed:
Merge pull request #10 from wordpress-mobile/upstream-2.2.4

Update `trunk` with upstream version `2.2.4`

fluiddot committed:
Merge pull request #11 from wordpress-mobile/wp-fork-2.2.4

Update wp-fork to version `2.2.4`

fluiddot committed:
Merge branch 'trunk' into wp-fork

fluiddot commented:
After bumping the version to `2.2.4`, future updates of the React Native version on the fork will require the following steps:
1. Update the RN version in the podspec.
https://github.com/wordpress-mobile/react-native-reanimated/blob/f1cd9fdc3f7fc216ccbc913b1ee1a5ab36110f83/RNReanimated.podspec#L5
1. Update the RN version in `android-npm/build.gradle` (Android project for integration).
https://github.com/wordpress-mobile/react-native-reanimated/blob/f1cd9fdc3f7fc216ccbc913b1ee1a5ab36110f83/android-npm/build.gradle#L8
1. Update the RN version in dependencies section of `android/build.gradle` (Android library project).
https://github.com/wordpress-mobile/react-native-reanimated/blob/f1cd9fdc3f7fc216ccbc913b1ee1a5ab36110f83/android/build.gradle#L447
1. Update the RN version in the script for creating the NPM package.
https://github.com/wordpress-mobile/react-native-reanimated/blob/f1cd9fdc3f7fc216ccbc913b1ee1a5ab36110f83/createNPMPackage.sh#L10-L11
1. Update the RN version in `package.json`.
https://github.com/wordpress-mobile/react-native-reanimated/blob/f1cd9fdc3f7fc216ccbc913b1ee1a5ab36110f83/package.json#L103
1. Run the command `./createNPMPackage.sh`.
**It will generate the AAR (Android library binary) and the tarball, these files should be pushed to the repository.**

**NOTES:**
- `./createNPMPackage.sh` scripts uses NDK for building the Android binaries. Please check that have installed the following Android SDK tools before running it (this can be verified in `Android Studio/Tools/SDK Manager/SDK Tools`):
    - NDK version `21.4.7075529`
    - CMake version `3.10.2`
- `./createNPMPackage.sh` script requires `wget` to be installed (you can install it, in case you don't have it, by running the command `brew install wget`).

fluiddot committed:
Merge branch 'upstream-2.4.1' into update-version-2.4.1

# Conflicts:
#	.github/workflows/main.yml
#	Common/cpp/Tools/JSIStoreValueUser.cpp
#	Common/cpp/headers/Tools/JSIStoreValueUser.h
#	Example/android/app/src/main/java/com/swmansion/reanimated/example/MainApplication.java
#	Example/android/build.gradle
#	Example/android/gradle.properties
#	Example/android/gradle/wrapper/gradle-wrapper.properties
#	Example/android/settings.gradle
#	Example/ios/Podfile.lock
#	Example/package.json
#	Example/yarn.lock
#	RNReanimated.podspec
#	android-npm/build.gradle
#	android/build.gradle
#	android/gradle/wrapper/gradle-wrapper.jar
#	android/gradle/wrapper/gradle-wrapper.properties
#	android/gradlew
#	android/gradlew.bat
#	createNPMPackage.sh
#	ios/native/NativeProxy.mm
#	ios/native/REAInitializer.h
#	ios/native/REAInitializer.mm
#	ios/native/UIResponder+Reanimated.mm
#	package.json
#	src/reanimated2/js-reanimated/index.web.ts
#	yarn.lock

fluiddot committed:
Fix merge conflicts

fluiddot committed:
Update yarn.lock

fluiddot committed:
Generate AAR file

fluiddot committed:
Generate tarball

fluiddot committed:
Merge pull request #13 from wordpress-mobile/update-version-2.4.1

Update `wp-fork` to version `2.4.1`

fluiddot committed:
Merge branch 'trunk' into upstream-2.4.1

# Conflicts:
#	.github/workflows/main.yml
#	Common/cpp/Tools/JSIStoreValueUser.cpp
#	Common/cpp/headers/Tools/JSIStoreValueUser.h
#	Example/android/app/src/main/java/com/swmansion/reanimated/example/MainApplication.java
#	Example/android/build.gradle
#	Example/android/gradle.properties
#	Example/android/gradle/wrapper/gradle-wrapper.properties
#	Example/android/settings.gradle
#	Example/ios/Podfile.lock
#	Example/package.json
#	Example/yarn.lock
#	RNReanimated.podspec
#	android-npm/build.gradle
#	android/build.gradle
#	createNPMPackage.sh
#	ios/native/NativeProxy.mm
#	ios/native/REAInitializer.h
#	ios/native/REAInitializer.mm
#	ios/native/UIResponder+Reanimated.mm
#	package.json
#	src/reanimated2/js-reanimated/index.web.ts
#	yarn.lock

fluiddot committed:
Merge pull request #12 from wordpress-mobile/upstream-2.4.1

Update `trunk` with upstream version `2.4.1`

fluiddot committed:
Merge branch 'trunk' into wp-fork

# Conflicts:
#	RNReanimated.podspec
#	android-npm/build.gradle
#	android/build.gradle
#	createNPMPackage.sh
#	package.json
#	yarn.lock

jacobarvidsson committed:
fix: android layout reanimation null pointer exception (2959) (#3162)

## Description
ReactApplicationContext.getCurentContext sometimes return null. This causes a crash (NullPointerException) in reanimated due to not checking if context exists. As per react native's examples, we can only measure device metrics when current activity is not null (https://github.com/facebook/react-native/blob/8bd3edec88148d0ab1f225d2119435681fbbba33/ReactCommon/react/nativemodule/samples/platform/android/SampleTurboModule.java#L56)

Fixes #2959.

## Changes

Check if current activity is not null before measuring the device.

fluiddot committed:
Update aar

fluiddot committed:
Bump version to wp-2

fluiddot committed:
Update tarball

fluiddot committed:
Merge pull request #15 from wordpress-mobile/fix-android-layout-reanimation-crash

Fix crash related to Android layout reanimation due to `NullPointerException`

Commit:
Invalidate display link instead of pausing it when invalidating the nodes manager

Pausing the display link on invalidation will still make it call the onAnimationFrame selector one more time as its already schedule. When this happens animation blocks will be called and crash while accessing the jsi runtime thats already dealloced.

fluiddot committed:
Bump package version

fluiddot committed:
Generate tarball

fluiddot committed:
Merge pull request #16 from wordpress-mobile/fix/invalidation-display-link

Invalidate display link instead of pausing it when invalidating the nodes manager

Commit:
Merge branch 'upstream-2.9.1' into wp-fork-2.9.1

# Conflicts:
#	RNReanimated.podspec
#	android-npm/build.gradle
#	android/build.gradle
#	android/gradle.properties
#	createNPMPackage.sh
#	package.json
#	yarn.lock

Commit:
Add React Native Reanimated 0.69 hermes binary

Commit:
Generate tarball

Commit:
Update tarball

fluiddot committed:
Upgrade compile and target sdk version to API 31

fluiddot committed:
Bump version to wp-4

Commit:
Remove previous version tarball

Commit:
Merge pull request #18 from wordpress-mobile/wp-fork-2.9.1

Update wp-fork to version `2.9.1`

Commit:
Merge pull request #17 from wordpress-mobile/upstream-2.9.1

Update `trunk` with upstream version `2.9.1`

fluiddot committed:
Merge branch 'trunk' into wp-fork

Commit:
Update tarball

Commit:
Merge pull request #20 from wordpress-mobile/wp-fork-2.9.1-update-tarball

Update tarball for `2.9.1`

fluiddot committed:
Merge pull request #19 from wordpress-mobile/update/android-12

Upgrade compile and target sdk version to Android API 31

fluiddot committed:
Merge branch 'wp-fork' into update-android-12-wp-fork

# Conflicts:
#	android-npm/expo/linking.gradle
#	android/build.gradle
#	package.json
#	react-native-reanimated-2.4.1-wp-4.tgz

Commit:
Updates createNPMPackage.sh to fix an issue after merge, it removes unneeded code comparing the changes from the original repo's file in 2.9.1.

fluiddot committed:
Merge pull request #22 from wordpress-mobile/fix/create-npm-package

Updates createNPMPackage.sh to fix an issue after merge

fluiddot committed:
Merge branch 'wp-fork' into update-android-12-wp-fork

fluiddot committed:
Bump version to wp-2

fluiddot committed:
Merge pull request #21 from wordpress-mobile/update-android-12-wp-fork

Upgrade compile and target sdk version to Android API 31 (version 2.9.1)

yusufyildirim committed:
Fix Jest 28+ `expect.extend is not a function` error (#3559)

Reanimated is not fully compatible with Jest 28+ even after the PR https://github.com/software-mansion/react-native-reanimated/pull/3347. Using it with Jest 28+ leads to this error:

<img width="796" alt="Ekran Resmi 2022-09-10 15 00 33" src="https://user-images.githubusercontent.com/22980987/189482337-cc2655f6-9463-4a05-9998-5fc3592f5cc0.png">

The reason is, the "expect" module's export approach has changed starting from Jest 28. It used to export the `expect` function directly, as can be seen [here](https://github.com/facebook/jest/blob/v27.5.1/packages/expect/src/index.ts#L441).

This means, requiring the "expect" module like this

```ts
const expect = require('expect')
```

was providing direct access to the `expect` function alongside with its attached methods, like `extend`.

This has [changed with Jest 28](https://github.com/facebook/jest/blob/v28.0.0/packages/expect/src/index.ts#L449). `export default expect` is the new export statement, which means requiring the "expect" package returns the module object, instead of the function.

https://github.com/software-mansion/react-native-reanimated/pull/3347 PR attempts to fix this error by assuming "expect" package should throw an error with Jest 28 and it can catch that and require the "expect" from the `@jest/globals` package for Jest 28+ project but it's not the case in reality. "expect" package is still shipped with Jest 28, so, requiring it doesn't throw an error as expected in the PR.

By knowing the export difference, I checked if the value of the module is object or not after requiring it. If it's an object, it means user is on Jest 28+ codebase. The rest is the same with the previous PR. I require the "expect" function from `@jest/globals` and use it.

Tested on Jest 29 and Jest 27.5.1

Fixes #3553 and #3215.

Commit:
Bump version to 2.9.1-wp-3

Commit:
Merge pull request #23 from wordpress-mobile/wp-fork-add-jest-28-support

Add support for Jest 28+

fluiddot committed:
Upgrade compile and target sdk version to API 33

fluiddot committed:
Bump version to wp-5

fluiddot committed:
Upgrade minimum sdk version to API 24

fluiddot committed:
Merge pull request #24 from wordpress-mobile/update/android-13

Upgrade compile and target sdk version to Android API 33
